src/main/webapp/app/view/invitation/accept-invitation.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnDestroy } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { InvitationService } from 'app/entities/invitation/invitation.service';
+import { filter, switchMap, takeUntil, tap } from 'rxjs/operators';
+import { AccountService } from 'app/core/auth/account.service';
+import { Subject } from 'rxjs';
+
+@Component({
+  selector: 'app-accept-invitation',
+  templateUrl: './accept-invitation.component.html',
+})
+export class AcceptInvitationComponent implements OnDestroy {
+  token?: string;
+  variant: 'login' | 'register';
+
+  private destroy$ = new Subject<void>();
+
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private invitationService: InvitationService,
+    private router: Router,
+    private accountService: AccountService
+  ) {
+    this.variant = activatedRoute.snapshot.data.variant;
+    this.activatedRoute.params
+      .pipe(
+        tap(params => (this.token = params['token'])),
+        switchMap(() => this.invitationService.checkTokenValidity(this.token!)),
+        tap(isTokenValid => {
+          if (!isTokenValid) {
+            this.router.navigate(['/invalid-token']);
+            return false;
+          }
+          return true;
+        }),
+        // if the token is valid and the user is already assigned in, automatically accept the invitation
+        filter(Boolean),
+        filter(() => this.accountService.isAuthenticated()),
+        switchMap(() => this.invitationService.assignCurrentUserToInvitation(this.token!)),
+        switchMap(() => this.accountService.identity(true)),
+        takeUntil(this.destroy$)
+      )
+      .subscribe(() => {
+        this.router.navigate(['/']);
+      });
+  }
+
+  ngOnDestroy(): void {
+    this.destroy$.next();
```

Comment:
dann noch `this.destroy$.complete()`.

Reply:
braucht es nicht unbedingt. das takeuntil funktioniert wenn das subject feuert. es muss nicht completen

Reply:
ja, das ist so. aber theoretisch ist das `Subject` dann immer noch offen..

Reply:
Ja aber da hört ja danach niemand drauf. das macht also nichts. ich wusste es auch nicht genau im detail aber dieser stackoverflow post erklärts gut https://stackoverflow.com/a/57009988

Reply:
ich weiss, dass es nicht nötig ist. habe den thread auch schon gelesen.